gammapy/cube/tests/test_ring.py

```diff
@@ -2,94 +2,109 @@
-    assert_allclose(result["alpha"].data[~in_fov], 0.0)
-
-
-class TestAdaptiveRingBackgroundEstimator:
-    def setup(self):
-        self.images = {}
-        self.images["counts"] = WcsNDMap.create(binsz=0.02, npix=101, dtype=float)
-        self.images["counts"].data += 1.0
-        self.images["background"] = WcsNDMap.create(binsz=0.02, npix=101, dtype=float)
-        self.images["background"].data += 1e10
-        exclusion = WcsNDMap.create(binsz=0.02, npix=101, dtype=float)
-        exclusion.data += 1
-        exclusion.data[40:60, 40:60] = 0
-        self.images["exclusion"] = exclusion
-
-    def test_run_const_width(self):
-        ring = AdaptiveRingBackgroundEstimator(
-            r_in=0.22 * u.deg, r_out_max=0.8 * u.deg, width=0.1 * u.deg
-        )
-        result = ring.run(self.images)
-
-        assert_allclose(result["background_ring"].data[50, 50], 1)
-        assert_allclose(result["alpha"].data[50, 50], 0.002638522427440632)
-        assert_allclose(result["exposure_off"].data[50, 50], 379 * 1e10)
-        assert_allclose(result["off"].data[50, 50], 379)
-
-        assert_allclose(result["background_ring"].data[0, 0], 1)
-        assert_allclose(result["alpha"].data[0, 0], 0.008928571428571418)
-        assert_allclose(result["exposure_off"].data[0, 0], 112 * 1e10)
-        assert_allclose(result["off"].data[0, 0], 112)
-
-    def test_run_const_r_in(self):
-        ring = AdaptiveRingBackgroundEstimator(
-            r_in=0.22 * u.deg,
-            r_out_max=0.8 * u.deg,
-            width=0.1 * u.deg,
-            method="fixed_r_in",
-        )
-        result = ring.run(self.images)
-
-        assert_allclose(result["background_ring"].data[50, 50], 1)
-        assert_allclose(result["alpha"].data[50, 50], 0.002638522427440632)
-        assert_allclose(result["exposure_off"].data[50, 50], 379 * 1e10)
-        assert_allclose(result["off"].data[50, 50], 379)
-
-        assert_allclose(result["background_ring"].data[0, 0], 1)
-        assert_allclose(result["alpha"].data[0, 0], 0.008928571428571418)
-        assert_allclose(result["exposure_off"].data[0, 0], 112 * 1e10)
-        assert_allclose(result["off"].data[0, 0], 112)
+from regions import CircleSkyRegion
+from astropy.coordinates import Angle, SkyCoord
+from gammapy.cube import AdaptiveRingBackgroundMaker, RingBackgroundMaker
+from gammapy.maps import WcsNDMap, WcsGeom, MapAxis
+from gammapy.cube.make import MapDatasetMaker
+from gammapy.utils.testing import requires_data
+from gammapy.data import DataStore
+
+
+@pytest.fixture(scope="session")
+def exclusion_region():
+    """Example mask for testing."""
+    pos = SkyCoord(83.633, 22.014, unit="deg", frame="icrs")
+    return CircleSkyRegion(pos, Angle(0.15, "deg"))
+
+
+@pytest.fixture(scope="session")
+def observations():
+    """Example observation list for testing."""
+    datastore = DataStore.from_file(
+        "$GAMMAPY_DATA/hess-dl3-dr1/hess-dl3-dr3-with-background.fits.gz"
```

Comment:
One more comment: this currently fails on Travis, because we removed the old background models. Just change this to `DataStore.from_dir("$GAMMAPY_DATA/hess-dl3-dr1")` and it should work again. Locally you have to update you gammapy data folder by deleting it and running `make dataset-download`.